test/mcs/test_defmcs.jl

```diff
@@ -92,7 +77,7 @@ end
 #
 global loop_counter = 0
 
-function outer_loop_func(sim::Simulation, tup)
+function outer_loop_func(sim_results::SimulationResults, tup)
```

Comment:
I can see use cases for loop funcs that use the trial data held in the `Simulation`. We might want to change the API to these to be `outer_loop_func(sim::Simulation, results::SimulationResults, tup)` now that we've split the original struct.